modules/nf-core/vamb/bin/main.nf

```diff
@@ -0,0 +1,83 @@
+process VAMB_BIN {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vamb:5.0.4--pyhdfd78af_0':
+        'biocontainers/vamb:5.0.4--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(assembly), path(abundance_tsv), path(bams, stageAs: "bams/*"), path(taxonomy)
+
+    output:
+    tuple val(meta), path("${prefix}.vamb/bins/*.fna.gz")             , emit: bins             , optional: true
+    tuple val(meta), path("${prefix}.vamb/vae*_clusters_metadata.tsv"), emit: clusters_metadata
+    tuple val(meta), path("${prefix}.vamb/vae*_clusters_split.tsv")   , emit: clusters_split   , optional: true
+    tuple val(meta), path("${prefix}.vamb/vae*_clusters_unsplit.tsv") , emit: clusters_unsplit
+    tuple val(meta), path("${prefix}.vamb/results_taxometer.tsv")     , emit: taxometer_results, optional: true
+    tuple val(meta), path("${prefix}.vamb/latent.npz")                , emit: latent_encoding  , optional: true
+    tuple val(meta), path("${prefix}.vamb/abundance.npz")             , emit: abundance
+    tuple val(meta), path("${prefix}.vamb/composition.npz")           , emit: composition
+    tuple val(meta), path("${prefix}.vamb/log.txt")                   , emit: log
+    path "versions.yml"                                               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    if(bams && abundance_tsv) {
+        error("ERROR: Both bams and abundance TSV supplied to Vamb!")
+    }
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def mode = taxonomy ? "taxvamb" : "default"
+    depth_input = abundance_tsv ? "--abundance_tsv ${abundance_tsv}" : "--bamdir bams/"
+    tax_input = taxonomy ? "--taxonomy ${taxonomy}" : ""
+    """
+    vamb bin \\
+        ${mode} \\
+        -p ${task.cpus} \\
+        --outdir ${prefix}.vamb/ \\
```

Comment:
Why are you setting `.vamb` on this?
If it is just convention, can we put it onto the default prefix, then it can be overridden if people want to do so.

Reply:
No convention specifically - I was just thinking about namespace collisions as the other metagenome binning module I'm working on also just spits out a directory with standardly-named files. But there's no particular reason this shouldn't just be `$prefix`, so I'll set it to that.

Reply:
Is there a potential for collision within the module? i.e. if one of the input files has the same name as the output files?

Reply:
I think that's very unlikely as the output that takes the prefix is a directory, but the inputs are a FASTA file and either a TSV or bam files.